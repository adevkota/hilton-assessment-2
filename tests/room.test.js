import React from "react";
import ReactDOM from "react-dom";
import Room from "../components/room";
import renderer from "react-test-renderer";

test("renders without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render((
		<Room
			selected={true} 
			index={0}
			selectable={false}
			numOfAdults={1} 
			numOfChildren={1} 
			roomSelected={()=>{}} 
			roomUnselected={()=>{}} 
			numOfAdultsChanged={()=>{}} 
			numOfChildrenChanged={()=>{}}
		/>
	), div);
	ReactDOM.unmountComponentAtNode(div);
});

test("matches snapshot", () => {
	const component = renderer.create(
		<Room
			selected={true} 
			index={1}
			selectable={true}
			numOfAdults={1} 
			numOfChildren={1} 
			roomSelected={()=>{}} 
			roomUnselected={()=>{}} 
			numOfAdultsChanged={()=>{}} 
			numOfChildrenChanged={()=>{}}
		/>
	);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
})