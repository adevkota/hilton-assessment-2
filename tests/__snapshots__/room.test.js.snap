// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="room__RoomWrapper-sc-14aj6pk-0 hJldwd"
  selected={true}
>
  <div
    className="room__RoomHeader-sc-14aj6pk-1 enkvyc"
    selected={true}
  >
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    <span>
      Room 
      2
    </span>
  </div>
  <div
    className="room__RoomContent-sc-14aj6pk-2 dFTFTL"
    selected={true}
  >
    <div
      className="room__HeadCountWrapper-sc-14aj6pk-3 dNzWIe"
    >
      <span>
        Adults
      </span>
      <span>
        (18+)
      </span>
      <select
        disabled={false}
        onChange={[Function]}
        value={1}
      >
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
      </select>
    </div>
    <div
      className="room__HeadCountWrapper-sc-14aj6pk-3 dNzWIe"
    >
      <span>
        Children
      </span>
      <span>
        (0-17)
      </span>
      <select
        disabled={false}
        onChange={[Function]}
        value={1}
      >
        <option
          value="0"
        >
          0
        </option>
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
      </select>
    </div>
  </div>
</div>
`;
